internal/resources/resource_lb_virtual_servers.go

```diff
@@ -15,87 +17,123 @@ import (
 func LoadBalancerVirtualServers() *schema.Resource {
 	return &schema.Resource{
 		Schema: map[string]*schema.Schema{
-			"vip_name": {
-				Type:        schema.TypeString,
+			"lb_id": {
+				Type:        schema.TypeInt,
 				Required:    true,
+				Description: "Parent lb ID, lb_id can be obtained by using LB datasource/resource.",
+				ForceNew:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
 				Description: "vip_name of Network loadbalancer virtual server name",
 			},
 			"description": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "description of Network loadbalancer virtual server",
-				ForceNew:    true,
 			},
 			"vip_address": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "vip_address of Network loadbalancer virtual server",
-				ForceNew:    true,
 			},
 			"vip_port": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "vip_port of network loadbalancer virtual server",
 			},
 			"pool": {
 				Type:        schema.TypeInt,
-				Required:    true,
+				Optional:    true,
 				Description: "pool of Network loadbalancer virtual server",
 			},
-			"ssl_cert": {
-				Type:        schema.TypeInt,
-				Required:    true,
-				Description: "ssl_cert of Network loadbalancer virtual server",
+			"vip_host_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "vip_host_name of Network loadbalancer virtual server",
+			},
+			"type": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validations.StringInSlice([]string{"http", "tcp", "udp"}, false),
+				Optional:         true,
+				Default:          "http",
+				Description:      "vip protocol of Network loadbalancer virtual server",
 			},
+			"tcp_application_profile":  schemas.TcpAppProfileSchema(),
+			"udp_application_profile":  schemas.UdpAppProfileSchema(),
+			"http_application_profile": schemas.HttpAppProfileSchema(),
+			"persistence": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validations.StringInSlice([]string{"SOURCE_IP", "COOKIE"}, false),
```

Comment:
Can it not have DISABLED ?

Reply:
in UI it is showing as DISABLED but in json for persistence is empty
Please refer the below screenshot
<img width="500" alt="ERR" src="https://user-images.githubusercontent.com/61727100/182768127-65803b85-11b0-4aca-804b-a59bf8cb03fd.PNG">

Reply:
but this field is marked compulsary, does it mean u need to pass some values ? make it optional then

Reply:
updated